lib/src/main/java/com/diffplug/spotless/kotlin/DiKTatStep.java

```diff
@@ -0,0 +1,168 @@
+		return create(versionDiktat, provisioner, true, userData, configPath);
+	}
+
+	public static FormatterStep create(String versionDiktat, Provisioner provisioner, boolean isScript, Map<String, String> userData, String configPath) {
+		Objects.requireNonNull(versionDiktat, "versionDiktat");
+		Objects.requireNonNull(provisioner, "provisioner");
+		return FormatterStep.createLazy(NAME,
+			() -> new DiKTatStep.State(versionDiktat, provisioner, isScript, userData, configPath),
+			DiKTatStep.State::createFormat);
+	}
+
+	static final class State implements Serializable {
+
+		private static final long serialVersionUID = 1L;
+
+		/** Are the files being linted Kotlin script files. */
+		private final boolean isScript;
+		private final String configPath;
+		private final String pkg;
+		private final String pkgKtlint;
+		/** The jar that contains the eclipse formatter. */
+		final JarState jar;
+		private final TreeMap<String, String> userData;
+
+		State(String versionDiktat, Provisioner provisioner, boolean isScript, Map<String, String> userData, String configPath) throws IOException {
+
+			HashSet<String> pkgSet = new HashSet<>();
+			pkgSet.add(MAVEN_COORDINATE + versionDiktat);
+
+			this.userData = new TreeMap<>(userData);
+			this.pkg = PACKAGE_DIKTAT;
+			this.pkgKtlint = PACKAGE_KTLINT;
+			this.jar = JarState.from(pkgSet, provisioner);
+			this.isScript = isScript;
+			this.configPath = configPath;
+		}
+
+		FormatterFunc createFormat() throws Exception {
+
+			ClassLoader classLoader = jar.getClassLoader();
+
+			// first, we get the diktat rules
+			if (configPath != null) {
+				System.setProperty("diktat.config.path", configPath);
+			}
+
+			Class<?> ruleSetProviderClass = classLoader.loadClass(pkg + ".ruleset.rules.DiktatRuleSetProvider");
+			Object standardRuleSet = ruleSetProviderClass.getMethod("get").invoke(ruleSetProviderClass.newInstance());
+			Iterable<?> ruleSets = Collections.singletonList(standardRuleSet);
+
+			// next, we create an error callback which throws an assertion error when the format is bad
+			Class<?> function2Interface = classLoader.loadClass("kotlin.jvm.functions.Function2");
+			Class<?> lintErrorClass = classLoader.loadClass(pkgKtlint + ".ktlint.core.LintError");
+			Method detailGetter = lintErrorClass.getMethod("getDetail");
+			Method lineGetter = lintErrorClass.getMethod("getLine");
+			Method colGetter = lintErrorClass.getMethod("getCol");
+			Object formatterCallback = Proxy.newProxyInstance(classLoader, new Class[]{function2Interface},
+				(proxy, method, args) -> {
+					Object lintError = args[0]; //ktlint.core.LintError
+					boolean corrected = (Boolean) args[1];
+					if (!corrected) {
+						String detail = (String) detailGetter.invoke(lintError);
+						int line = (Integer) lineGetter.invoke(lintError);
+						int col = (Integer) colGetter.invoke(lintError);
+						throw new AssertionError("Error on line: " + line + ", column: " + col + "\n" + detail);
```

Comment:
```suggestion
						throw new AssertionError("Error on line: " + line + ", column: " + col + " cannot be fixed automatically\n" + detail);
```
I feel like we usually have more errors that need manual intervention than ktlint. Maybe worth considering once again, how to handle this better.